lte/gateway/c/core/oai/tasks/amf/amf_app_ue_context.cpp

```diff
@@ -192,14 +346,40 @@ amf_context_t* amf_context_get(const amf_ue_ngap_id_t ue_id) {
  ***************************************************************************/
 ue_m5gmm_context_s* amf_ue_context_exists_amf_ue_ngap_id(
     const amf_ue_ngap_id_t amf_ue_ngap_id) {
-  std::unordered_map<amf_ue_ngap_id_t, ue_m5gmm_context_s*>::iterator
-      found_ue_id = ue_context_map.find(amf_ue_ngap_id);
+  struct ue_m5gmm_context_s* ue_context_p = NULL;
+  map_uint64_ue_context_t* amf_state_ue_id_ht = get_amf_ue_state();
 
-  if (found_ue_id == ue_context_map.end()) {
-    return NULL;
-  } else {
-    return found_ue_id->second;
+  if (amf_state_ue_id_ht->get(amf_ue_ngap_id, &ue_context_p) != magma::MAP_OK) {
+    OAILOG_WARNING(LOG_AMF_APP,
+                   " amf_ue_ngap_id " AMF_UE_NGAP_ID_FMT "does not exist\n",
+                   amf_ue_ngap_id);
+    return ue_context_p;
+  }
+
+  return ue_context_p;
+}
+
+/****************************************************************************
+ **                                                                        **
+ ** Name:    amf_ue_context_exists_gnb_ue_ngap_id()                        **
+ **                                                                        **
+ ** Description: Checks if UE context exists already or not based on       **
+ **              gnb_key                                                   **
+ **                                                                        **
+ **                                                                        **
+ ***************************************************************************/
+ue_m5gmm_context_s* amf_ue_context_exists_gnb_ue_ngap_id(
```

Comment:
Can be implemented something similar to amf_ue_context_exists_amf_ue_ngap_id() to decrease the number of lines

Reply:
Comment addressed.

Reply:
Comment addressed.

Reply:
Comment addressed.